### ISSUE TITLE
switch ElasticUtils to become a django layer on top of elasticsearch-dsl-py

### ISSUE THREAD

**Issue by willkg** [mozilla/elasticutils#252]
The Elasticsearch folks have started a library called elasticseach-dsl. The code is at https://github.com/elasticsearch/elasticsearch-dsl-py . @robhudson and I have already done some work on it.

Like ElasticUtils, elasticsearch-dsl-py sits on top of elasticsearch-py and provides a nicer search API. There are some differences between how that API works, but generally speaking, it solves the same problems we were solving with ElasticUtils.

My current thinking is that we rewrite ElasticUtils with the following requirements:
1. base it off of elasticsearch-dsl-py
2. return to being a Django-specific library (people can use elasticsearch-dsl-py for non-Django usage)
3. reduce the scope to providing bits that make Elasticsearch convenient in a Django context

Why? Well, there's no need to be a general purpose library anymore--elasticsearch-dsl-py will probably do that better. I think there is a need for a library that makes things convenient for Django. At a minimum, we'd use this at Mozilla. Maintenance of large libraries where the maintainers don't have a lot of time generally sucks. Reducing the focus/scope should make it easier to maintain at a higher quality. Plus, elasticsearch-dsl-py makes some different architectural decisions and thus already supports a large number of Elasticsearch features that will be difficult to do in ElasticUtils as it currently exists.

Things that need to happen before this can happen:
1. elasticsearch-dsl-py needs a critical mass of functionality to be able to base the new ElasticUtils on it
2. there needs to be an elasticsearch-dsl-py release--we don't want to base our library on alpha code

**Comment by adriaant** [comment]
Can you point me to some of those django shim libraries you mention?

**Comment by ChristopherRabotin** [comment]
Could one of those be https://github.com/Sparrho/bungiesearch? 

Disclosure: I am the maintainer of this project.

**Comment by ap0091** [comment]
@willkg , can you let us know what some of the other libraries are?

Thank You

**Comment by willkg** [comment]
@ap0091 Read the comments in this issue and do a google search. I don't have experience with any of them.
